src/Microsoft.Azure.Functions.Extensions.Mcp/Http/McpHttpUtility.cs

```diff
@@ -39,4 +42,82 @@ internal static void SetSseContext(HttpContext context)
 
         context.Features.GetRequiredFeature<IHttpResponseBodyFeature>().DisableBuffering();
     }
+
+    internal static async ValueTask<JsonRpcMessage?> ProcessJsonRpcPayloadAsync(HttpRequest request, JsonSerializerOptions options, CancellationToken cancellationToken, bool unwrapOnly = false)
+    {
+        // Process the incoming request body as JSON. Support both raw JSON-RPC messages and
+        // wrapped payloads with the shape: { "isFunctionsMcpResult": true, "content": <JSON-RPC message> }
+        // 
+        // When unwrapOnly is false: If the wrapper is present, deserialize the inner "content" as the JsonRpcMessage. 
+        // Otherwise, deserialize the root object directly as a JsonRpcMessage and return it.
+        //
+        // When unwrapOnly is true: If the wrapper is present, replace the request.Body stream with a memory stream 
+        // containing only the inner content. If no wrapper is present, leave the original body intact.
+
+        // If the body is empty, return null.
+        if (request.ContentLength == null || request.ContentLength == 0)
+        {
+            return null;
+        }
+
+        // Read the request body into a JsonDocument for inspection.
+        request.EnableBuffering();
+        try
+        {
+            using var doc = await JsonDocument.ParseAsync(request.Body, cancellationToken: cancellationToken);
+            var root = doc.RootElement;
+
+            JsonElement messageElement = root;
+            bool isWrapped = false;
+
+            if (root.ValueKind == JsonValueKind.Object &&
+                root.TryGetProperty("isFunctionsMcpResult", out var marker) &&
+                marker.ValueKind == JsonValueKind.True &&
+                root.TryGetProperty("content", out var content))
+            {
+                messageElement = content;
+                isWrapped = true;
+            }
+
+            if (unwrapOnly)
+            {
+                if (isWrapped)
+                {
+                    var inner = messageElement.GetRawText();
+                    var bytes = System.Text.Encoding.UTF8.GetBytes(inner);
+                    request.Body = new MemoryStream(bytes);
+                    request.ContentLength = bytes.Length;
+                }
+                else
+                {
+                    // Reset position so downstream readers can consume the original body.
+                    request.Body.Seek(0, SeekOrigin.Begin);
+                }
+                return null;
+            }
+            else
+            {
+                var raw = messageElement.GetRawText();
+                return JsonSerializer.Deserialize<JsonRpcMessage>(raw, options);
```

Comment:
What are we doing on json exceptions? And do we know that messageElement is json?